spec/features/support/providers/locations/creating_multiple_locations_spec.rb

```diff
@@ -16,36 +16,120 @@
+    when_i_am_redirected_to_the_locations_page
+    and_i_see_the_text_two_locations_added
+    then_the_database_should_have_updated_with_the_new_locations
+  end
+
+  scenario 'clicking back' do
+    given_the_multiple_locations_feature_flag_is_active
+    and_i_visit_the_multiple_locations_new_page
+    and_i_submit_the_form_with_two_locations
+    and_i_see_the_text_one_of_two
+    and_i_should_see_that_the_text_field_has_been_prepopulated('Name', 'Tottenham')
+
+    and_i_submit_a_valid_form
+    and_i_see_the_text_two_of_two
+    and_i_see_that_the_text_field_has_been_prepopulated('Name', 'Tottenham Hotspur')
+    and_i_submit_a_valid_form
+    and_i_am_redirected_to_the_multiple_location_confirm_page
+    and_the_database_should_not_have_updated_with_the_new_location
+
+    when_i_click_back
+    then_i_see_the_text_two_of_two
+    and_i_see_that_the_text_field_has_been_prepopulated('Name', 'Tottenham Hotspur')
+    and_i_click_back
+    and_i_see_the_text_one_of_two
+    and_i_should_see_that_the_text_field_has_been_prepopulated('Name', 'Tottenham')
+
+    and_i_click_back
+    and_i_am_on_the_new_multiple_locations_page
+
+    and_i_click_back
+
+    and_i_should_be_on_the_provider_locations_page
+  end
+
+  scenario 'cancel from multiple locations new page' do
+    given_the_multiple_locations_feature_flag_is_active
+    and_i_visit_the_multiple_locations_new_page
+
+    when_i_click_cancel
+    then_i_should_be_on_the_provider_locations_page
+  end
+
+  scenario 'cancel from multiple locations new page for a location' do
+    given_the_multiple_locations_feature_flag_is_active
+    and_i_visit_the_multiple_locations_new_page
+    and_i_submit_the_form_with_two_locations
+    and_i_see_the_text_one_of_two
+    and_i_should_see_that_the_text_field_has_been_prepopulated('Name', 'Tottenham')
+
+    when_i_click_cancel
+    then_i_should_be_on_the_provider_locations_page
+  end
+
+  scenario 'cancel from multiple location confirm page' do
+    given_the_multiple_locations_feature_flag_is_active
+    and_i_visit_the_multiple_locations_new_page
+    and_i_submit_the_form_with_two_locations
+    and_i_submit_a_valid_form
+    and_i_see_the_text_two_of_two
+    and_i_see_that_the_text_field_has_been_prepopulated('Name', 'Tottenham Hotspur')
+    and_i_submit_a_valid_form
+    and_i_am_redirected_to_the_multiple_location_confirm_page
+
+    when_i_click_cancel
+    then_i_should_be_on_the_provider_locations_page
   end
 
   scenario 'feature flag off' do
     when_i_visit_a_provider_locations_page
     then_i_should_not_see_the_add_multiple_locations_link
   end
 
-  def then_i_should_be_redirected_to_the_provider_location_page
-    expect(page).to have_current_path support_recruitment_cycle_provider_locations_path(recruitment_cycle_year: Settings.current_recruitment_cycle_year, provider_id: provider.id)
+  def then_the_database_should_have_updated_with_the_new_locations
+    expect(Site.find_by(location_name: 'Tottenham').present?).to be true
```

Comment:
```suggestion
expect(Site.exists?(location_name: 'Tottenham')).to be true
```

and the ones below

Reply:
Done: https://github.com/DFE-Digital/publish-teacher-training/pull/3420/commits/023a1e28ca10fc296dc52ae1892bf3ecc13b18a0